src/library/scala/concurrent/ExecutionContext.scala

```diff
@@ -139,9 +139,30 @@ object ExecutionContext {
    *
    * @return the global `ExecutionContext`
    */
-  def global: ExecutionContextExecutor = Implicits.global.asInstanceOf[ExecutionContextExecutor]
+  final def global: ExecutionContextExecutor = Implicits.global.asInstanceOf[ExecutionContextExecutor]
```

Comment:
I'm having trouble understanding why this reads from `Implicits.global` and does a cast, rather than `Implicits.global` reading from this and not needing to cast

Reply:
@NthPortal Great question! It's because Implicits.global needs to be typed `ExecutionContext` and not `ExecutionContextExecutor` otherwise its more specific type would take precedence over other implicit `ExecutionContext`s in scope. The non-implicit `ExecutionContext.global` can be typed as `ExecutionContextExecutor` so it can be used for Java APIs requiring `Executor`.

Reply:
@viktorklang what I meant was, why not instead do this?

```scala
lazy final val global: ExecutionContextExecutor = impl.ExecutionContextImpl.fromExecutor(null: Executor)
object Implicits {
  implicit final def global: ExecutionContext = ExecutionContext.global // no cast needed!
}
```

Reply:
@NthPortal Done!